packages/webdriverio/src/commands/browser/switchFrame.ts

```diff
@@ -49,73 +68,211 @@ export async function switchFrame (
+            throw new Error('Cannot use a string to fetch a context in WebDriver Classic')
+        }
+        return switchToFrame(this, context)
+    }
+
     /**
      * if context is `null` the user is switching to the top level frame
      * which is always represented by the value of `getWindowHandle`
      */
     if (context === null) {
         const handle = await this.getWindowHandle()
-        await switchToFrameHelper(this, handle)
+        switchToFrameHelper(this, handle)
+        await switchToFrame(this, context)
         return handle
     }
 
+    /**
+     * if context is a string the user is trying to switch to a frame
+     * that has a url that matches the string or a context id that matches
+     * the string.
+     */
     if (typeof context === 'string') {
         const tree = await this.browsingContextGetTree({})
-        const urlContext = findContext(context, tree.contexts, byUrl)?.context
+        let newContextId: string | undefined
+
+        const urlContext = (
+            findContext(context, tree.contexts, byUrl) ||
+            /**
+             * In case the user provides an url without `/` at the end, e.g. `https://example.com`,
+             * the `browsingContextGetTree` command may return a context with the url `https://example.com/`.
+             */
+            findContext(`${context}/`, tree.contexts, byUrl)
+        )
+        const urlContextContaining = findContext(context, tree.contexts, byUrlContaining)
+        const contextIdContext = findContext(context, tree.contexts, byContextId)
         if (urlContext) {
-            await switchToFrameHelper(this, urlContext)
-            return urlContext
+            log.info(`Found context by url "${urlContext.url}" with context id "${urlContext.context}"`)
+            newContextId = urlContext.context
+        } else if (urlContextContaining) {
+            log.info(`Found context by url containing "${urlContextContaining.url}" with context id "${urlContextContaining.context}"`)
+            newContextId = urlContextContaining.context
+        } else if (contextIdContext) {
+            log.info(`Found context by id "${contextIdContext}" with url "${contextIdContext.url}"`)
+            newContextId = contextIdContext.context
+        }
+
+        if (!newContextId) {
+            throw new Error(`No frame with url or id "${context}" found!`)
+        }
+
+        const sessionContext = getContextManager(this)
+        const currentContext = await sessionContext.getCurrentContext()
+        const allContexts = await getFlatContextTree(this)
+
+        /**
+         * Fetch all iframes located in any available frame
+         */
+        const allFrames = (await Promise.all(Object.keys(allContexts).map(async (id) => {
+            /**
+             * first, fetch all iframes in given browsing context
+             */
+            const { nodes } = await this.browsingContextLocateNodes({
+                locator: { type: 'css', value: 'iframe' },
```

Comment:
Does this also work for classic frames: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/frame?